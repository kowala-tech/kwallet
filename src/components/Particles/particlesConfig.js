const config = {
	"particles": {
		"number": {
			"value": 25,
			"density": {
				"enable": true,
				"value_area": 300
			}
		},
		"color": {
			"value": "#ffffff"
		},
		"shape": {
			"type": "circle"
		},
		"opacity": {
			"value": 0.5,
			"random": true,
			"anim": {
				"enable": true,
				"speed": 1,
				"opacity_min": 0.1,
				"sync": false
			}
		},
		"size": {
			"value": 2,
			"random": true,
			"anim": {
				"enable": true,
				"speed": 4.87155337598649,
				"size_min": 0.1,
				"sync": false
			}
		},
		"line_linked": {
			"enable": true,
			"distance": 64,
			"color": "#ffffff",
			"opacity": 0.05,
			"width": 1
		},
		"move": {
			"enable": true,
			"speed": 0,
			"direction": "none",
			"random": true,
			"straight": false,
			"out_mode": "out",
			"bounce": false,
			"attract": {
				"enable": false,
				"rotateX": 600,
				"rotateY": 1200
			}
		}
	},
	"interactivity": {
		"detect_on": "canvas",
		"events": {
			"onhover": {
				"enable": false,
				"mode": "push"
			},
			"onclick": {
				"enable": false,
				"mode": "push"
			},
			"resize": false
		},
		"modes": {
			"grab": {
				"distance": 400,
				"line_linked": {
					"opacity": 1
				}
			},
			"bubble": {
				"distance": 400,
				"size": 40,
				"duration": 2,
				"opacity": 8,
				"speed": 3
			},
			"repulse": {
				"distance": 200,
				"duration": 0.4
			},
			"push": {
				"particles_nb": 4
			},
			"remove": {
				"particles_nb": 2
			}
		}
	},
	"retina_detect": true
};

export default config;
