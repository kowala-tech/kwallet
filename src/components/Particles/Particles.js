import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
// Component Related Imports
import styles from "./styles";
import ParticlesJS from "react-particles-js";
import particlesConfig from "./particlesConfig";

class Particles extends React.Component {

	render() {
		const {
			classes
		} = this.props;

		return (
			<ParticlesJS
				className={classes.particles}
				params={particlesConfig}
			/>
		);
	}
}

Particles.propTypes = {
	classes: PropTypes.object.isRequired
};

export default compose(
	withStyles(styles)
)(Particles);
