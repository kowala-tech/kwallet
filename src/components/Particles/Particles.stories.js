import React from "react";
import { storiesOf } from "@storybook/react";
// Story related component imports
import Particles from "./Particles";

const Container = (storyFn) => (
	<div style={{
		backgroundColor: "#1f033d",
		width: "100%",
		height: "100%",
		position: "fixed"
	}}>
		{ storyFn() }
	</div>
);

storiesOf("Particles", module)
	.addDecorator(Container)
	.add("default", () =>
		(<Particles />)
	);
