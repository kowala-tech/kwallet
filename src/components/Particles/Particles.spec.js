import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import Particles from "./Particles";

describe("Particles", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<Particles/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
