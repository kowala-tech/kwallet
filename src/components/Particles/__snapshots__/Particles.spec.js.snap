// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Particles it should match the snapshot 1`] = `
<Particles
  classes={
    Object {
      "particles": "Particles-particles-1",
    }
  }
/>
`;
